src/services/pool.ts

```diff
@@ -109,9 +143,42 @@ const mapBranches = (branches: BranchEntry[], fullBaseUrl: string) => {
   }));
 };
 
-const mapRepos = (repo: RepoDocument): RepoResponse => ({
-  repoName: repo.repoName,
-  project: repo.project,
-  search: repo.search,
-  branches: mapBranches(repo.branches, getRepoUrl(repo.url[ENV_URL_KEY], repo.prefix[ENV_URL_KEY])),
-});
+const mapRepos = async (repo: RepoDocument) => {
+  try {
+    const query: Filter<DocsetDocument> = { project: repo.project };
+    const strictOptions: FindOptions = {
+      projection: {
+        project: 1,
+        url: 1,
+        prefix: 1,
+      },
+    };
+
+    const repo_docset = await db
+      .collection<DocsetDocument>(DOCSETS_COLLECTION)
+      .find(query, strictOptions)
```

Comment:
(Maybe nit) Since we only want the first in the array, and since there should be a 1-many relationship between docsets and repos branches, I think it might be safe to just make this a `findOne` (unless we ever need to handle any logic that stems from finding more than 1 result)